tests/test_tps_regression.rs

```diff
@@ -26,48 +26,26 @@ mod test {
         submitted_transactions: Arc<RwLock<Vec<(u64, Instant)>>>,
     ) {
         let (money_sender, money_receiver) = sample_two_nodes(nodes.len());
-        let tx_receiver = sample_queryable_node(&nodes);
+        let tx_receiver = money_sender;
         // Update nonces.
         let mut nonces = nonces.write().unwrap();
         nonces[money_sender] += 1;
         let nonce = nonces[money_sender];
 
         let sender_acc = nodes[money_sender].read().unwrap().account_id().unwrap();
         let receiver_acc = nodes[money_receiver].read().unwrap().account_id().unwrap();
-        let transaction = SignedTransaction::send_money(
-            nonce,
-            sender_acc,
-            receiver_acc,
-            &*nodes[money_sender].read().unwrap().signer(),
-            1,
-            nodes[money_sender].read().unwrap().user().get_best_block_hash().unwrap(),
-        );
-        nodes[tx_receiver].read().unwrap().add_transaction(transaction).unwrap();
-        submitted_transactions.write().unwrap().push((1, Instant::now()));
-    }
-
-    /// Iterates over the given records of transactions, bucketizes them, and computes the tps.
-    /// Can be used for debugging this test.
-    fn bucketize_tps(
-        recorded_transactions: &Arc<RwLock<Vec<(u64, Instant)>>>,
-        bucket_size: Duration,
-    ) -> Vec<u64> {
-        let mut bucket_start = recorded_transactions.read().unwrap()[0].1;
-        let mut buckets = vec![0u64];
-        for b in &*recorded_transactions.read().unwrap() {
-            if bucket_start + bucket_size >= b.1 {
-                *buckets.last_mut().unwrap() += b.0;
-            } else {
-                *buckets.last_mut().unwrap() /= bucket_size.as_secs() as u64;
-                buckets.push(b.0);
-                bucket_start += Duration::from_nanos(
-                    ((b.1 - bucket_start).as_nanos() as u64) / (bucket_size.as_nanos() as u64)
-                        * (bucket_size.as_nanos() as u64),
-                );
-            }
+        if let Some(block_hash) = nodes[tx_receiver].read().unwrap().user().get_best_block_hash() {
+            let transaction = SignedTransaction::send_money(
+                nonce,
+                sender_acc,
+                receiver_acc,
+                &*nodes[money_sender].read().unwrap().signer(),
+                1,
+                block_hash,
+            );
+            nodes[tx_receiver].read().unwrap().add_transaction(transaction).unwrap();
+            submitted_transactions.write().unwrap().push((1, Instant::now()));
```

Comment:
Since there is no bucketing we can just remove the timestamp from the collection.